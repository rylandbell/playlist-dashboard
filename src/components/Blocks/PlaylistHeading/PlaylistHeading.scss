@import 'variables';

.playlist-heading__title {
  color: white;
  margin-top: 5px;
  margin-right: 20px;
  margin-bottom: 0;
}

.playlist-heading__tracks-count {
  font-size: 12px;
  color: $gray-light;
  margin-left: 2px;
}

.playlist-heading__image {
  width: 48px;
  display: inline-block;
  vertical-align: top;
  margin-right: 10px;
  margin-top: 4px;
}

.playlist-heading__text {
  display: inline-block;
}
