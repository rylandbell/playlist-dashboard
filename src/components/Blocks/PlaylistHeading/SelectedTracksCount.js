import React from 'react';
import { getPlaylistDuration } from '../../../helper.js';

const SelectedTracksCount = ({ tracks, filteredTracks }) => {
  //only count tracks with non-null features data
  const allTracksCount = tracks.filter(track => !!track).length;
  const filteredTracksCount = filteredTracks.length;
  const duration = getPlaylistDuration(filteredTracks);

  let durationPhrase = '';
  if (duration.hr && duration.hr > 0) {
    durationPhrase += `${duration.hr} hr `;
  }
  durationPhrase += `${duration.min} min`;

  return (
    <div>
      <span className="playlist-heading__tracks-count">{`${filteredTracksCount} of ${allTracksCount} songs, ${durationPhrase}`}</span>
    </div>
  );
};

export default SelectedTracksCount;
