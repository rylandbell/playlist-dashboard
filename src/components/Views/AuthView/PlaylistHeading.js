import React, { Component } from 'react';

class PlaylistHeading extends Component {
  render() {
    const userId = this.props.reduxState.userId;
    const playlistOwnerId = this.props.reduxState.selectedPlaylist.owner.id;
    const idMatch = userId === playlistOwnerId;
    return (
      <h3 className="tracks__heading"> {this.props.reduxState.selectedPlaylist.name}
        {idMatch ? null: <small>&nbsp;by {playlistOwnerId}</small>}
      </h3>
    );
  }
}

export default PlaylistHeading;

