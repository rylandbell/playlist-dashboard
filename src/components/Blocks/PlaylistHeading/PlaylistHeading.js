import React from 'react';
import { connect } from 'react-redux';

import SelectedTracksCount from './SelectedTracksCount';

import './PlaylistHeading.css';

const mapStateToProps = (state) => {
  return {
    userId: state.auth.userId,
    selectedPlaylist: state.selectedPlaylist,
    tracks: state.tracks
  }
}

const mapDispatchToProps = (dispatch) => {
  return {}
}

const PlaylistHeading = ({userId, selectedPlaylist, filteredTracks, tracks}) => {
  const playlistOwnerId = selectedPlaylist.ownerId;
  const idMatch = userId === playlistOwnerId;
  return (
    <div>
      <h3 className="playlist-heading__title"> {selectedPlaylist.name}
        {idMatch ? null: <small>&nbsp;by {playlistOwnerId}</small>}
      </h3>
      <SelectedTracksCount tracks={tracks} filteredTracks={filteredTracks} />
    </div>
  );
}

const PlaylistHeadingContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(PlaylistHeading)

export default PlaylistHeadingContainer;

