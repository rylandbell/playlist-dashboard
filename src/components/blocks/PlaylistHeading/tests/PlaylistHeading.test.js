import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import {PlaylistHeading} from "../PlaylistHeading";
import {testFilteredTracks, testTracks, testSelectedPlaylist} from './test-data';

describe('PlaylistHeading component', () => {
  const testProps = {
  userId: 'abc',
  selectedPlaylist: testSelectedPlaylist,
  filteredTracks: testFilteredTracks,
  tracks: testTracks
}
  it('renders without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<PlaylistHeading {...testProps} />, div);
  });

  it('shallow render matches snapshot', () => {
    const tree = toJson(shallow(<PlaylistHeading {...testProps} />));
    expect(tree).toMatchSnapshot();
  });
});