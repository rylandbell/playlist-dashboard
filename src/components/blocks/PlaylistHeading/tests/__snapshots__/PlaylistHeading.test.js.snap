// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaylistHeading component shallow render matches snapshot 1`] = `
<div>
  <img
    alt="playlist cover"
    className="playlist-heading__image hidden-xs"
    src="https://i.scdn.co/image/cfb2948cf2e897460b0feasdfaef1e29d6d8"
  />
  <div
    className="playlist-heading__text"
  >
    <h3
      className="playlist-heading__title"
    >
       
      Bruce Springsteen – Hammersmith Odeon, London '75
      <small>
         by 
        1237555400
      </small>
    </h3>
    <SelectedTracksCount
      filteredTracks={
        Array [
          Object {
            "acousticness": 0.575,
            "artist": "KMD",
            "danceability": 0.881,
            "duration_ms": 24162633,
            "energy": 0.836,
            "id": "6UUY9L3jGnK507sKWK20s3",
            "instrumentalness": 0,
            "liveness": 0.105,
            "name": "Peachfuzz",
            "popularity": 0.35,
            "uri": "spotify:track:6UUY9L3jGnK507sKWK20s3",
            "valence": 0.461,
          },
          Object {
            "acousticness": 0.046,
            "artist": "Tidus",
            "danceability": 0.939,
            "duration_ms": 192000,
            "energy": 0.653,
            "id": "0tJWc3RXOuli1FjEgzpCn7",
            "instrumentalness": 0.277,
            "liveness": 0.0336,
            "name": "Delancey",
            "popularity": 0.41,
            "uri": "spotify:track:0tJWc3RXOuli1FjEgzpCn7",
            "valence": 0.81,
          },
        ]
      }
      tracks={
        Array [
          Object {
            "acousticness": 0.575,
            "artist": "KMD",
            "danceability": 0.881,
            "duration_ms": 241626,
            "energy": 0.836,
            "id": "6UUY9L3jGnK507sKWK20s3",
            "instrumentalness": 0,
            "liveness": 0.105,
            "name": "Peachfuzz",
            "popularity": 0.35,
            "uri": "spotify:track:6UUY9L3jGnK507sKWK20s3",
            "valence": 0.461,
          },
          Object {
            "acousticness": 0.046,
            "artist": "Tidus",
            "danceability": 0.939,
            "duration_ms": 192000,
            "energy": 0.653,
            "id": "0tJWc3RXOuli1FjEgzpCn7",
            "instrumentalness": 0.277,
            "liveness": 0.0336,
            "name": "Delancey",
            "popularity": 0.41,
            "uri": "spotify:track:0tJWc3RXOuli1FjEgzpCn7",
            "valence": 0.81,
          },
          Object {
            "acousticness": 0.857,
            "artist": "Bruce Springsteen",
            "danceability": 0.476,
            "duration_ms": 350480,
            "energy": 0.47,
            "id": "2tFykPvhvN1AdUkeQQLCuV",
            "instrumentalness": 0,
            "liveness": 0.976,
            "name": "Thunder Road - The Song - Live at Hammersmith Odeon Audio",
            "popularity": 0.39,
            "uri": "spotify:track:2tFykPvhvN1AdUkeQQLCuV",
            "valence": 0.353,
          },
        ]
      }
    />
  </div>
</div>
`;
